### PR TITLE
Added custom context support in installer

### PR DESCRIPTION
**Pull request by devansh42**
Added custom context support in the installer, user can now download only required adapters

Signed-off-by: Devansh Gupta <devanshguptamrt@gmail.com>

**Description**
* This add up functionality to pull only required adapters
* User can give a space sperated list of adapters to pull in current context
* If didn't give any list then it would fallback to default context, and would pull all the available adapters
**Available adapter options are**
* cpx
* consul
* istio
* linkerd
* nsm
* kuma
* cpx
* osm
* traefik

This PR fixes [this](https://github.com/layer5io/service-mesh-labs/issues/9) issue

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by devansh42** [comment]
Hey, @leecalcote can you review this?

File: meshery [hunk 1]
```diff
@@ -127,42 +127,77 @@ unzip ${PWD}/meshery.zip -d ${PWD}/$temporary_dir_name
 install_dir=${MESHERY_INSTALL_DIR}
 if [ -z "$install_dir" ]
 then
-    install_dir="/usr/local/bin"
+install_dir="/usr/local/bin"
```
Comment (leecalcote) [inline meshery:130]: @devansh42 these outdentations (if I can make up that word) seem undesirable given that these statements are executed _if_ the preceding line is true. Could you have a look at reverting this formatting?

**Review by navendu-pottekkat — COMMENTED** [review]
I believe that we need to change the one liner to as mentioned [here](https://github.com/meshery/meshery/discussions/3168) for this to work.

Invoking `mesheryctl system context create` without a config file would first create a local context and set it as the `current-context`. So creating a new `custom` context without the `--set` flag would not work.

If we are allowing users to change the context during initialisation, then we can change the name of the initial context from `local` to `default`.

File: meshery (hunk 1 above)
Comment (devansh42) [reply meshery:130]: I'd surely add these changes to the next commit.

File: meshery [hunk 2]
```diff
@@ -127,42 +127,77 @@ unzip ${PWD}/meshery.zip -d ${PWD}/$temporary_dir_name
...
+        traefik)
+            adapterString="$adapterString -a meshery-$adapter-mesh "
+        ;;
+        *)
+            # We are just neglecting invalid mesh options
+        ;;
+    esac    
+  done
+
+    mesheryctl system context create custom -s $adapterString --yes
```
Comment (devansh42) [inline meshery:198]: @navendu-pottekkat  **-s** is ensuring that we switched to custom context and **--yes** ensures that all the prompts are satisfied.

**Review by devansh42 — COMMENTED** [review]
> If we are allowing users to change the context during initialisation, then we can change the name of the initial context from `local` to `default`.

Sure, semantically it makes more sense.

File: meshery (hunk 1 above)
Comment (leecalcote) [reply meshery:130]: Ok, signal when done.

**Comment by leecalcote** [comment]
@hexxdump will you review this in context of your deploy argument?

**Review by navendu-pottekkat — COMMENTED** [review]
@devansh42 the new changes are from line 168 right? Could you revert the other changes?

File: meshery [hunk 3]
```diff
@@ -127,42 +127,77 @@ unzip ${PWD}/meshery.zip -d ${PWD}/$temporary_dir_name
...
     exit 1
-  fi
+fi
 fi
 
 
 printf "Removing installation files and opening Meshery..."
 rm -rf meshery.zip ${PWD}/$temporary_dir_name/
 
+
```
Comment (hexxdump) [inline meshery:168]: @devansh42, You should move these changes towards the end of the script. i.e. after the recently introduced 
 if [ "$DEPLOY_MESHERY" == false ] ; check.

File: meshery [hunk 4]
```diff
@@ -17,27 +17,27 @@
...
-            ;;
-        kubernetes)
-            PLATFORM="kubernetes"
-            ;;
-        *)
-            echo "Invalid platform: $OPTARG"
-            exit 1
-            ;;
-        esac
+        p)
```
Comment (hexxdump) [inline meshery:20]: rollback these indentation changes, we do not require them

File: meshery [hunk 5]
```diff
@@ -70,9 +70,9 @@ OSARCHITECTURE="$(uname -m)"
 OS="$(uname)"
 
 if [ "x${OS}" = "xDarwin" ] ; then
-  OSEXT="Darwin"
+    OSEXT="Darwin"
```
Comment (hexxdump) [inline meshery:73]: looks good

**Comment by hexxdump** [comment]
@devansh42, pls resolve merge conflicts. we are good to merge these changes.

File: meshery [hunk 6]
```diff
@@ -186,5 +186,38 @@ then
...
+# osm
+# traefik
+
+## Any adapter other than above would be ignored
+if [ ! -z "$ADAPTERS" ]; then
+    adpaterString=
+    for adapter in $ADAPTERS; do
+        case $adapter in
+            cpx|consul|istio|linkerd|nsm|kuma|cpx|osm)
+                adapterString="$adapterString -a meshery-$adapter "
```
Comment (leecalcote) [inline meshery:208]: @devansh42 will you add nginx, app-mesh, and tanzu-sm?

**Review by hexxdump — APPROVED** [review]
LGTM

**Review by navendu-pottekkat — APPROVED** [review]
Thank you @devansh42 💯

**Pull request merged**